gnd/src/main/java/com/google/android/gnd/MainViewModel.java

```diff
@@ -149,9 +154,18 @@ void onSignedOut() {
     navigator.navigate(SignInFragmentDirections.showSignInScreen());
   }
 
+  /**
+   * Called when the user signs in as well as when the previous authentication state is restored
+   * when reopening the app.
+   */
   private void onSignedIn() {
     hideProgressDialog();
-    navigator.navigate(HomeScreenFragmentDirections.showHomeScreen());
+
+    if (localValueStore.areTermsAccepted()) {
```

Comment:
It looks like terms of service dialog is invoked here regardless of whether there are terms in the db.

Let's also check this through the terms manager instead of injecting localValueStore directly into the vm. 

You could add a method there like "areTermsOfServiceRequired()" and "areTermsOfServiceAccepted()" that we check here.

Reply:
In #845 have injected localValueStore in TermsOfServiceRepository https://github.com/google/ground-android/blob/b6b97d600b74ccc312455d109dd185261f5e4701/gnd/src/main/java/com/google/android/gnd/repository/TermsOfServiceRepository.java